### PR TITLE
Use `sql_function!` for `count`

### PR DESCRIPTION
**Pull request by sgrif**
This change technically allows more than we did before, but it doesn't
feel like a major problem to me.

### CODE REVIEW

File: diesel/src/expression/mod.rs [hunk 1]
```diff
@@ -47,9 +47,12 @@ pub mod sql_literal;
 pub mod subselect;
 
 #[doc(hidden)]
+#[allow(non_camel_case_types)]
 pub mod dsl {
+    use dsl::SqlTypeOf;
+
     #[doc(inline)]
-    pub use super::count::{count, count_star};
+    pub use super::count::*;
```
Comment (weiznich) [inline diesel/src/expression/mod.rs]: pub using items with `#[doc(inline)]` does not actually show the documentation in that place (Rustdoc will only render `pub use super::count::*;` here not the actual imported types.

Comment (sgrif) [reply diesel/src/expression/mod.rs]: I don't think that's correct. http://docs.diesel.rs/diesel/dsl/index.html https://github.com/diesel-rs/diesel/blob/master/diesel/src/lib.rs#L187-L191

Comment (weiznich) [reply diesel/src/expression/mod.rs]: Hmm, then this only works inside of an crate. (There was [this PR](https://github.com/diesel-rs/diesel/pull/1410) for diesel_migrations)

Comment (sgrif) [reply diesel/src/expression/mod.rs]: Correct, you cannot inline documentation across crates.

**Pull request merged**